### PR TITLE
Add a link to the RN 0.62 upgrade guide

### PR DESCRIPTION
**Pull request by matt-oakes**
# Summary

Added a link to the [React Native 0.62 upgrade guide](https://reactnative.thenativebits.com/courses/upgrading-react-native/upgrade-to-react-native-0.62/). This is in the same style as the guides for previous versions, which are also linked in the upgrade helper. It tries to explain *why* the changes are needed, rather than just having a list of changes to make.

## Test Plan

N/A

## What are the steps to reproduce?

N/A

## Checklist

- [X] I tested this thoroughly
- [ ] I added the documentation in `README.md` (if needed)

### CODE REVIEW

File: src/releases/0.62.js [hunk 1]
```diff
@@ -10,6 +10,11 @@ export default {
           'Official blog post about the major changes on React Native 0.62',
         url: 'https://reactnative.dev/blog/2020/03/26/version-0.62'
       },
+      {
+        title: 'Tutorial on upgrading to React Native 0.62',
```
[the remark this answers is not in the document]
Comment (kelset) [reply src/releases/0.62.js:14]: I think it should be `[External] Tutorial on upgrading to React Native 0.62`

**Comment by kelset** [comment]
Aside from the change in title, I'm ok with having this merged.

There are similar links to Matt's blogposts for previous versions - and while I share your concerns @pvinis I also think that as long as it's clearly an external link, that would be enough.

What I would like to see moving on though, @matt-oakes, is for you to collaborate directly in the Upgrade Support and having your guides made in the same way as Pavlos is suggesting - it's the classic problem of writing your own blogpost on something VS making a PR to the documentation and I wish at least for the RNC we could all collaborate more.

**Comment by pvinis** [comment]
Having you, Matt, help and make this kind of posts on the upgrade-support repo would be amazing!

**Comment by matt-oakes** [comment]
I've made the change to the title, including going back and adding `[External]` to the existing blog post links 👍 

> What I would like to see moving on though, @matt-oakes, is for you to collaborate directly in the Upgrade Support and having your guides made in the same way as Pavlos is suggesting - it's the classic problem of writing your own blogpost on something VS making a PR to the documentation and I wish at least for the RNC we could all collaborate more.

Yes. I'm planning to get more involved with helping to answer questions over there. I actually missed that it existed until late last week as I was focused on getting a large project completed.

Given that the support repo now exists, I was actually in two minds about if I should write the blog post at all. In the end, I did because a long form guide didn't seem to fit with the "issues" format that the repo is going for. Creating a permanently opened issue with an upgrade guide inside it didn't seem like a good fit.

To me, it seems like the help needs to be split into two parts:

1. A series of useful guides to help with upgrading, which people can contribute to through pull requests.
2. An area where people can ask for support and get specific answers. This can then act as a reference in itself, or the answers can be folded into the upgrade guides, maybe just as link to "common issues & solutions".

Personally, I'm not convinced that Github issues are a good place for people to ask for solutions. It tends to devolve into a load of comments saying "me 2!!!!11!!" and "any updates?". This is exactly why Stack Overflow was created as it allows the accepted answers to bubble to the top and naturally removes the noise.

Either way, I'd be happy to help out with the upgrade support 👍 

> If anyone has any suggestions on how to make contributing to the Upgrade Support better / more appealing, I'm more than happy to hear them!

I think the best way to encourage people to contribute to official documentation rather than write blog posts is to allow for their name to be listed on it. You can see an example of this at the bottom of the new testing guide on the React Native website:

https://reactnative.dev/docs/next/testing-overview#links

This should be updated to include anyone who contributes.

> I feel it's a bit not great to shove this on a random user's face, specifically when are linked from an "official" tool to an external blog. I'm all for getting paid for one's effort, don't get me wrong, and I am guilty of having a link on the purge repo for a buymecoffee or whatever it's called, but it's a tiny link in the readme that has only been clicked twice, one was a friend even. And to be fair, I'm happy to remove this from there too.

Understood. I didn't mean for these guides to be seen as "official" in any way, and while they are partly written with promotion in mind (as most blog posts are) they are also intended to help people out. As I said above, I'd be happy to contribute to the support repo, I just wasn't sure how this sort of a guide fitted into it yet.

**Comment by lucasbento** [comment]
> A series of useful guides to help with upgrading, which people can contribute to through pull requests.

I like this, any suggestions on how we could achieve it? Perhaps having a `.md` file for each major release in the `upgrade-support` repo?

> An area where people can ask for support and get specific answers. This can then act as a reference in itself, or the answers can be folded into the upgrade guides, maybe just as link to "common issues & solutions".

I'm getting in touch with @github to have the new Discussions feature enabled for `upgrade-support`, I'm not sure how that's gonna go as it's not in public beta yet but it does exactly what you said with questions & answers.

> This should be updated to include anyone who contributes.

Completely agree and there's already an issue at facebook/react-native-website#1670 to track this.

**Comment by matt-oakes** [comment]
> I'm getting in touch with @github to have the new Discussions feature enabled for upgrade-support, I'm not sure how that's gonna go as it's not in public beta yet but it does exactly what you said with questions & answers.

I didn't even know that was a thing! It looks perfect 👍 

> Completely agree and there's already an issue at facebook/react-native-website#1670 to track this.

Nice 👍 

> I like this, any suggestions on how we could achieve it? Perhaps having a .md file for each major release in the upgrade-support repo?

It could either be markdown files which are just linked from the README of the repo, or it could be a Github Pages website if you want to get fancy.

Should we create a dedicated discussion for this topic? Is it best for it to live in the upgrade support repo itself, or in the "discussions & proposals" one?

**Comment by lucasbento** [comment]
> Should we create a dedicated discussion for this topic? Is it best for it to live in the upgrade support repo itself, or in the "discussions & proposals" one?

@kelset any thoughts on this? on D&P it will have more visibility which can be a double-edged sword.

Btw I have scheduled a meeting with @github for April 21st to talk about the possibility of adding `Discussions` to `upgrade-support`.

---

@pvinis @kelset: can we merge this PR?

**Comment by pvinis** [comment]
Just as a final check, add a 👍 on this comment and we merge. If there is anything else we can do here, then please comment. :)

**Comment by TheSavior** [comment]
Thank you all for working on this and planning to do this for more releases going forward! I think this will be super helpful for complex Xcode changes until e have an improved method.

**Comment by pvinis** [comment]
Strap in folks :D

**Comment by lucasbento** [comment]
Created an issue at react-native-community/upgrade-support#61, please drop in ideas there :)

**Pull request merged**